### PR TITLE
Complete refresh

### LINKED ISSUE #35: create_private_room & add_user
I don't see an option to create_private_room or add_user to a private room.  Is that something that can be added easily?

### LINKED ISSUE #38: Update README.md to support all implemented calls

### LINKED ISSUE #46: In delete_user and get_user_info methods, add support for username from just user_id
The user_id is hashed and impossible to get without logging in from the API and as the user to get it.

For both of these methods, the API supports username=, which I feel is much more convenient for API calls rather than the hashed name.

### LINKED ISSUE #48: Attachments in message payload
**Is your feature request related to a problem? Please describe.**
Hi, I ran into a problem that I could not find how to send an attachment, which is described by the endpoint in api rocketchat [here](https://docs.rocket.chat/guides/developer-guides/rest-api/chat/postmessage#attachments-detail)

**Describe the solution you'd like**
Implement the necessary keys from the api documentation, which are described [here](https://docs.rocket.chat/guides/developer-guides/rest-api/chat/postmessage#payload) in the send_message method with the correct payload build

**Describe alternatives you've considered**
I looked at the methods that are in the api class. File upload implemented, I saw issue #22 on this
However, I did not find the possibility of creating my own attachment

P.S.
I just started using this module, so if this feature is available, then I apologize, but I think it will be good if it is described explicitly here.
If it is not implemented yet, then I will probably be able to someday make a PR for this, if I use this module for my integrations. And when I figure out the code and structure.

In any case, I want to hear your comment about this feature.

Thank you!

### PR DESCRIPTION
**Pull request by soundstorm**
Allow call to get ID of public room (channel) or user; bump to 1.2.15
Will allow better code for https://github.com/fraschetti/Octoslack/pull/93
Closes #46 as getting the id is now possible (but still not without auth).
Allow to create private rooms, invite users to rooms/channels (closes #35)
Complete documentation in README (closes #38 and closes #48)

### CODE REVIEW

**Comment by soundstorm** [comment]
Although the call of `get_room_id` (and `set_room_topic` too) to acquire the id of a group instead of a channel is misleading, as `get_room_history` is a call on channels, but would be a breaking change otherwise.
Maybe this could be a cleanup for further releases?
Or add an alias for them like `get_private_room_id`, `set_private_room_topic`, `get_public_room_history` and deprecate the others.

**Comment by soundstorm** [comment]
Hey @pipozzz can you have a look? Closed most of your open issues and made a few enhancements.

**Comment by pipozzz** [comment]
I'm a little busy right now, but definitely I'll review it and let you know. Thanks for contribution.

File: setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 
 setup(name='rocket-python',
-      version='1.2.14',
+      version='1.2.15',
```
Comment (pipozzz) [inline setup.py:5]: Please do not touch version numbers, I update it automatically wen I'm doing releases.
```suggestion
      version='1.2.14',
```

File: .bumpversion.cfg [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 1.2.14
+current_version = 1.2.15
```
Comment (pipozzz) [inline .bumpversion.cfg:2]: Please do not touch version numbers, I update it automatically wen I'm doing releases.
```suggestion
current_version = 1.2.14
```

**Review by pipozzz — CHANGES_REQUESTED** [review]

**Review by pipozzz — CHANGES_REQUESTED** [review]

**Comment by pipozzz** [comment]
Also we need to resolve failed build before merging, btw Thanks for your contribution again ! 👍

**Comment by GrabsOrion** [comment]
Hello, @soundstorm! 
Are you planning to fix the failed build?
As much as I know, the message from Travis CI not so crucial: 
> 'rocketchat.calls.groups.upload_remote_file.UploadRemoteFile' imported but unused

I'm very new to programming, but interested in the project, thank you!

**Comment by pipozzz** [comment]
It seems that code is OK just test coverage dropped below 85%.

**Comment by soundstorm** [comment]
This is still in the refresh branch after two years, will there be a release with those changes anytime?

**Comment by pipozzz** [comment]
Hi @soundstorm, I was very busy, but Yes I'll review it and merge it if it's fine. There was failed some test last time I took a look. So let me check it again and merge it if it's fine. Thanks

**Pull request merged**